PassingData/SecondViewController.swift

```diff
@@ -9,26 +9,43 @@
 import UIKit
 
 class SecondViewController: UIViewController {
-
-    @IBOutlet weak var incomingtextlabel: UILabel!
     
-    var incomingText : String?
+    @IBOutlet weak var userName: UITextField!
+    
+    @IBOutlet weak var email_id: UITextField!
+    
+    @IBOutlet weak var password: UITextField!
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        incomingtextlabel.text = incomingText
-        // Do any additional setup after loading the view.
     }
+    
+    func emailValidation(email:String) -> Bool {
+        let predicate = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,4}"
+        
+        let emailTest = NSPredicate(format:"SELF MATCHES %@", predicate)
+        return emailTest.evaluate(with: email)
+    }
+    
+    let storyboar : UIStoryboard =  UIStoryboard(name: "Main", bundle: nil)
+    
+    @IBAction func signup(_ sender: Any) {
+        
+        let isemailValidated = emailValidation(email: email_id.text!)
+        
+        let nextViewController = self.storyboar.instantiateViewController(withIdentifier: "WelcomeViewController") as! WelcomeViewController
```

Comment:
Try to reduce the use of force unwrapping. Try using guard let and if let. This page has lots of ! change it.